File: Androidparty/app/src/main/java/com/tzemaitis/androidparty/ServersRecyclerViewAdapter.java
package com.tzemaitis.androidparty;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class ServersRecyclerViewAdapter extends RecyclerView.Adapter<ServersRecyclerViewAdapter.ViewHolder> {

    private final List<Server> mValues;

    public ServersRecyclerViewAdapter(List<Server> items) {
        mValues = items;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.server_list_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        holder.mItem = mValues.get(position);
        holder.mNameView.setText(mValues.get(position).getName());
        holder.mDistanceView.setText(mValues.get(position).getDistance() + " km");
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final View mView;
        public final TextView mNameView;
        public final TextView mDistanceView;
        public Server mItem;

        public ViewHolder(View view) {
            super(view);
            mView = view;
            mNameView = (TextView) view.findViewById(R.id.server_name);
            mDistanceView = (TextView) view.findViewById(R.id.distance);
        }
    }
}

File: app/src/main/java/com/assignment/android_party2/servers/ui/adapter/ServersRecyclerViewAdapter.kt
package com.assignment.android_party2.servers.ui.adapter

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.assignment.android_party2.R
import com.assignment.android_party2.servers.models.ServerModel

class ServersRecyclerViewAdapter(private val context: Activity, private val servers: List<ServerModel>) :
    RecyclerView.Adapter<ServersRecyclerViewAdapter.ServerViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerViewHolder {
        val rootView = LayoutInflater.from(context).inflate(R.layout.server_item, parent, false)
        return ServerViewHolder(rootView)
    }

    override fun onBindViewHolder(holder: ServerViewHolder, position: Int) {
        val server = servers[position]
        holder.name.text = server.name
        holder.distance.text = server.distance.toString()+ "  km"
    }

    override fun getItemCount(): Int {
        return servers.size
    }

    inner class ServerViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var name: TextView = itemView.findViewById(R.id.name)
        var distance: TextView = itemView.findViewById(R.id.distance)

    }
}
